### PR TITLE
fix(component-header): fix dataLayer firing on parent dropdown click

### PR DESCRIPTION
**Pull request by davidornelas11**
### Description
fix dataLayer firing on parent dropdown click

### FOR APPROVERS

- [Percy build approval](https://percy.io/5eae92d9/-all-UDS-packages)

### CODE REVIEW

**Review by chill5-0 — DISMISSED** [review]
Approved

JIRA: https://asudev.jira.com/browse/WS2-2353

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
Data Layer is firing for the dropdowns, top level menu items, secondary menu items.

But we have mostly empty events firing for...
* Links in the gray universal nav area (search input and toggle controls are okay for that, though.)
  * change onFocus() to onClick() and add more detail to `DEFAUL_GA_EVENT`, e.g. region should be `navbar`
* ASU Logo
* Partner Logo
* Site title (subdomain)
* Parent Unit
* CTAs in the top level of the header
* Buttons in the dropdowns
... at least, I'm not able to confirm in Storybook.  Let's do more work to verify.

**Comment by mlsamuelson** [comment]
For reference https://xd.adobe.com/view/907fadbe-e90a-40a8-a0b7-2396bd9450dd-c52c/

File: packages/component-header/src/components/HeaderMain/Logo/index.js
Comment (chill5-0) [inline packages/component-header/src/components/HeaderMain/Logo/index.js:1]: @davidornelas11 I think this should be onClick, not onFocus?

Comment (davidornelas11) [reply packages/component-header/src/components/HeaderMain/Logo/index.js:1]: Good catch. Fixed

**Review by chill5-0 — DISMISSED** [review]
Approved

File: packages/component-header/src/components/HeaderMain/Title/index.js [hunk 1]
```diff
@@ -37,15 +37,35 @@ const Title = () => {
...
+          }
           title={`${parentOrg} home page`}
         >
           {parentOrg}
         </a>
         <a
           className={`subunit-name ${active ? "active" : ""}`}
           href={baseUrl}
-          onFocus={() => trackGAEvent({ text: title })}
+          onFocus={() =>
```
Comment (mlsamuelson) [inline packages/component-header/src/components/HeaderMain/Title/index.js]: Should this one still be onFocus() ?

**Review by mlsamuelson — APPROVED** [review]

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
3 build errors to fix:

Screenshot 2024-10-03 at 9 49 08 AM

File: packages/component-header/src/components/HeaderMain/NavbarContainer/DropdownItem/index.js [hunk 2]
```diff
@@ -54,7 +53,7 @@ const LinkItem = ({ link, dropdownName, handleLinkEvent }) => (
       href={link.href}
       onClick={e => handleLinkEvent(e, link)}
       onKeyDown={e => handleLinkEvent(e, link)}
-      onFocus={() => trackGAEvent({ text: link.text, component: dropdownName })}
+      onClick={() => trackGAEvent({ text: link.text, component: dropdownName })}
```
Comment (mlsamuelson) [inline packages/component-header/src/components/HeaderMain/NavbarContainer/DropdownItem/index.js]: onClick on lines 54 and 56 - triggers no duplicate props linting error. Need just one. Assuming it's 56.

File: packages/component-header/types/main.d.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export interface Button {
   text: string;
   classes?: string;
   onClick?: () => void;
-  onFocus?: () => void;
+  onClick?: () => void;
```
Comment (mlsamuelson) [inline packages/component-header/types/main.d.ts:17]: Duplicates previous line.

**Review by mlsamuelson — APPROVED** [review]
Verified.

**Comment by mlsamuelson** [comment]
Associated ticket for this is WS2-2366

**Pull request merged**